hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPOfferService.java

```diff
@@ -324,6 +325,10 @@ private void notifyNamenodeBlock(ExtendedBlock block, BlockStatus status,
     final ReceivedDeletedBlockInfo info = new ReceivedDeletedBlockInfo(
         block.getLocalBlock(), status, delHint);
     final DatanodeStorage storage = dn.getFSDataset().getStorage(storageUuid);
+    if (storage == null) {
```

Comment:
Thanks @kokonguyen191 for your work.

there is a small question. when the execution reaches this, can we directly record `dnMetrics.incrNullStorageBlockReports()` and return?

Reply:
Good idea. Better to stop it here rather than let the code propagate further down. I'll change it later once I charge my laptop.

Reply:
Done